nsxt/resource_nsxt_policy_predefined_security_policy.go

```diff
@@ -0,0 +1,439 @@
+	logLabel := ""
+	rule.Tag = &logLabel
+
+	return rule
+}
+
+func createChildDomainWithSecurityPolicy(domain string, policyID string, policy model.SecurityPolicy) (*data.StructValue, error) {
+	converter := bindings.NewTypeConverter()
+	converter.SetMode(bindings.REST)
+
+	childPolicy := model.ChildSecurityPolicy{
+		ResourceType:   "ChildSecurityPolicy",
+		SecurityPolicy: &policy,
+	}
+
+	dataValue, errors := converter.ConvertToVapi(childPolicy, model.ChildSecurityPolicyBindingType())
+	if len(errors) > 0 {
+		return nil, errors[0]
+	}
+
+	var domainChildren []*data.StructValue
+	domainChildren = append(domainChildren, dataValue.(*data.StructValue))
+
+	targetType := "Domain"
+	childDomain := model.ChildResourceReference{
+		Id:           &domain,
+		ResourceType: "ChildResourceReference",
+		TargetType:   &targetType,
+		Children:     domainChildren,
+	}
+
+	dataValue, errors = converter.ConvertToVapi(childDomain, model.ChildResourceReferenceBindingType())
+	if len(errors) > 0 {
+		return nil, errors[0]
+	}
+	return dataValue.(*data.StructValue), nil
+}
+
+func updatePolicyPredefinedSecurityPolicy(id string, d *schema.ResourceData, m interface{}) error {
+
+	connector := getPolicyConnector(m)
+	path := d.Get("path").(string)
+	domain := getDomainFromResourcePath(path)
+
+	if domain == "" {
+		return fmt.Errorf("Failed to extract domain from Security Policy path %s", path)
+	}
+
+	predefinedPolicy, err := getSecurityPolicyInDomain(id, domain, connector, isPolicyGlobalManager(m))
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("description") {
+		description := d.Get("description").(string)
+		predefinedPolicy.Description = &description
+	}
+
+	if d.HasChange("tag") {
+		predefinedPolicy.Tags = getPolicyTagsFromSchema(d)
+	}
+
+	var childRules []*data.StructValue
+	if d.HasChange("rule") {
+		oldRules, _ := d.GetChange("rule")
+		rules := getPolicyRulesFromSchema(d, true)
+
+		existingRules := make(map[string]bool)
+		for _, rule := range rules {
+			ruleID := newUUID()
+			if rule.Id != nil {
+				ruleID = *rule.Id
+				existingRules[ruleID] = true
+			} else {
+				rule.Id = &ruleID
+			}
+
+			childRule, err := createPolicyChildRule(ruleID, rule, false)
+			if err != nil {
+				return err
+			}
+			log.Printf("[DEBUG]: Adding child rule with id %s", ruleID)
+			childRules = append(childRules, childRule)
+		}
+
+		for _, oldRule := range oldRules.([]interface{}) {
+			oldRuleMap := oldRule.(map[string]interface{})
+			oldRuleID := oldRuleMap["nsx_id"].(string)
+			if _, exists := existingRules[oldRuleID]; !exists {
+				resourceType := "Rule"
+				rule := model.Rule{
+					Id:           &oldRuleID,
+					ResourceType: &resourceType,
+				}
+
+				childRule, err := createPolicyChildRule(oldRuleID, rule, true)
+				if err != nil {
+					return err
+				}
+				log.Printf("[DEBUG]: Deleting child rule with id %s", oldRuleID)
+				childRules = append(childRules, childRule)
+
+			}
+		}
+	}
+
+	if d.HasChange("default_rule") {
```

Comment:
Maybe also using "else if" here since the two "if"s are mutually exclusive?

Reply:
In this resource you can define both rules and default rules.

Reply:
Ah I see. Sorry for missing this.